### PR TITLE
Explain routing in router docs

### PR DESCRIPTION
**Pull request by jayjun**
This pull request adds a “Routing” section to `Phoenix.Router` docs.

While Rails is a popular framework, the routing convention mimicked by Phoenix should be documented for first-time developers and those from other stacks (where regex is far more common). Features like globbing also don’t behave like Rails or Unix counterparts.

### CODE REVIEW

**Comment by chrismccord** [comment]
❤️❤️❤️🐥🔥

File: lib/phoenix/router.ex [hunk 1]
```diff
@@ -28,17 +28,72 @@ defmodule Phoenix.Router do
...
+
+      get "/pages", PageController, :index
+
+  matches a `GET` request to `/pages` and dispatches it to the `index` action in
+  `PageController`.
+
+      get "/pages/:page", PageController, :show
+
+  matches `/pages/hello` and dispatches to the `show` action with
+  `%{"path" => "hello"}` in `params`.
```
Comment (jarroput) [inline lib/phoenix/router.ex]: @jayjun Should it not match `%{"page" => "hello"}` in `params`, as `:page` is the param name?

Comment (Gazler) [reply lib/phoenix/router.ex]: @jarroput Good spot, feel free to open a PR.

Comment (jayjun) [reply lib/phoenix/router.ex]: Good catch @jarroput! Fix in #3526.

**Pull request merged**